driver/pace/driver/state.py

```diff
@@ -175,12 +179,116 @@ def _overwrite_state_from_restart(
     return state
 
 
+def _overwrite_state_from_fortran_restart(
+    path: str,
+    communicator: pace.util.CubedSphereCommunicator,
+    state: Union[fv3core.DycoreState, pace.physics.PhysicsState, TendencyState],
+    restart_file_prefix: str,
+    is_gpu_backend: bool,
+):
+    """
+    Args:
+        path: path to restart files
+        rank: current rank number
+        state: an empty state
+        restart_file_prefix: file prefix name to read
+
+    Returns:
+        state: new state filled with restart files
+    """
+
+    state_dict = _driver_state_to_dict(state, restart_file_prefix, is_gpu_backend)
+
+    state_dict = pace.util.open_restart(path, communicator, fortran_restart=True)
+
+    state = _dict_state_to_driver_state(state_dict, state, is_gpu_backend)
+
+    return state
+
+
+def _driver_state_to_dict(
+    driver_state: Union[fv3core.DycoreState, pace.physics.PhysicsState, TendencyState],
+    restart_file_prefix: str,
+    is_gpu_backend: bool,
+):
+    """
+    Takes a Pace driver state
+    and returns a dict of state quantities with their Fortran names
+    """
+
+    dict_state = {}
+
+    for _field in fields(type(driver_state)):
```

Comment:
Instead of doing this dynamically, it would be much safer to have a manual list of field names to iterate over.

Reply:
This is also a good idea.